dvc/remote/ssh/__init__.py

```diff
@@ -128,20 +132,37 @@ def ssh(self, path_info):
             password=self.password,
         )
 
-    def exists(self, path_infos):
-        single_path = False
-
-        if not isinstance(path_infos, list):
-            single_path = True
-            path_infos = [path_infos]
+    def exists(self, path_info):
+        with self.ssh(path_info) as ssh:
+            return ssh.exists(path_info.path)
+
+    def batch_exists(self, path_infos, callback):
+        def _exists(chunk_and_channel):
+            chunk, channel = chunk_and_channel
+            ret = []
+            for path in chunk:
+                try:
+                    channel.stat(path)
+                    ret.append(True)
```

Comment:
Why do we ignore the result of `channel.stat()` here? We don't in `.exists()`.

Reply:
There's no such thing as `channel.exists` and we can't call `self` / `ssh` exists because SSH is not thread safe (using the same connection in different threads).

Reply:
I don't understand the answer. The channel is an sftp instance, so why we don't examine what `.stat()` returns the same way we do for single exists?

Reply:
I don't know why I answered that, @Suor, it doesn't make sense at all! my bad :sweat_smile: 

We are ignoring the result because we only care about knowing whether the file exists or not.